test/common/http/http2/codec_impl_test.cc

```diff
@@ -236,8 +245,13 @@ TEST_P(Http2CodecImplTest, DeferredReset) {
   server_wrapper_.dispatch(Buffer::OwnedImpl(), server_);
 }
 
-INSTANTIATE_TEST_CASE_P(Http2CodecImplTest, Http2CodecImplTest,
-                        testing::Values(0, CodecOptions::NoCompression));
+INSTANTIATE_TEST_CASE_P(
+    Http2CodecImplTest, Http2CodecImplTest,
+    testing::Values(Http2SettingsTestParam(0, Http2Settings::DEFAULT_MAX_CONCURRENT_STREAMS,
+                                           Http2Settings::DEFAULT_INITIAL_WINDOW_SIZE),
+                    Http2SettingsTestParam(Http2Settings::CodecOptions::DisableDynamicHPACKTable,
+                                           Http2Settings::DEFAULT_MAX_CONCURRENT_STREAMS,
+                                           Http2Settings::DEFAULT_INITIAL_WINDOW_SIZE)));
```

Comment:
You could expand the testing by also parameterizing this test case with the edge cases of valid values for max_concurrent_streams and default_initial_window_size.